contracts/BatchExchange.sol

```diff
@@ -293,6 +293,10 @@ contract BatchExchange is EpochTokenLocker {
         require(tokenIdsForPrice[0] != 0, "Fee token has fixed price!");
         require(tokenIdsForPrice.checkPriceOrdering(), "prices are not ordered by tokenId");
         require(owners.length <= MAX_TOUCHED_ORDERS, "Solution exceeds MAX_TOUCHED_ORDERS");
+        // Further assumptions are: owners.length == orderIds.length && owners.length == buyVolumes.length
+        // && prices.length == tokenIdsForPrice.length
+        // These assumptions are not checked explicitly, as violations of these constraints can not be used
```

Comment:
Do we have tests trying to exploit an out of bounds situation?

Reply:
Good question! I think we should. I may have included one or two of these at some point for other functions, but possibly not for `submitSolution`.

Ahhh yes... `placeOrders` has tests with array index out of bound errors.